src/Rings/mpoly-ideals.jl

```diff
@@ -48,10 +48,12 @@ julia> typeof(J)
 MPolyIdeal{MPolyElem_dec{fmpq, fmpq_mpoly}}
 ```
 """
-function ideal(g::Vector{T}) where {T <: MPolyElem}
-  @assert length(g) > 0
-  @assert all(x->parent(x) == parent(g[1]), g)
-  return MPolyIdeal(g)
+function ideal(R::MPolyRing, g::Vector)
+  h = elem_type(R)[R(f) for f = g]
```

Comment:
@simonbrandhorst pointed out the other day that it might be desirable to *only* accept elements as input which are already elements of `R` and throw an error otherwise. I disagreed, but couldn't point out another instance where implicit coercion was done. Now there's one here which is not due to me! 

Question: Do we have it written down somewhere which convention we want to follow for this? Just so we do not stumble upon this discussion again and again?

Reply:
Yes, we do this in lots of different constructs. For example:
```
julia> matrix(QQ, 2, 2, [1, 2, 3, 4])
[1   2]
[3   4]
```
Note that this is mainly convenience for interactive use.

I think this is orthogonal to this PR here.